### ISSUE TITLE
KeyError: 'task_url' when trying to upload the tests

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy-bot-old#64]
I just got this error when it tried to upload the tests:

``` pytb
Traceback (most recent call last):
  File "./sympy-bot", line 244, in <module>
    main()
  File "./sympy-bot", line 96, in main
    review(n, options)
  File "./sympy-bot", line 226, in review
    report_url = reviews_sympy_org_upload(data, url_base)
  File "/Users/aaronmeurer/Documents/python/sympy/sympy-bot/utils.py", line 153, in reviews_sympy_org_upload
    return r["task_url"]
KeyError: 'task_url'
```

**Comment by certik** [comment]
Thanks for the bug report. This means that the online app responded with a valid json (!) but without the task_url, so there must be some error on the server. So this part of sympy-bot needs to be robustified, so that it retries again.

**Comment by certik** [comment]
Ok --- so Google has changed it's free quota and now our app engine is over quota for datastore writes.... So I am decreasing update intervals to see if it helps, and then we can gradually increase them again.

**Comment by certik** [comment]
So Google has disabled all write operations for our webapp, until the end of the billing period, which will be reset in 6 hours from now. I have drastically reduced the update frequency, so let's see if we run over the quota tomorrow. If we don't, then I can increase the frequency a bit. Also, we should implement a hook in our app to update on newly created pull requests immediately. Also I think I am writing into the datastore even if the content didn't change --- just fixing this little thing should allow us to run very frequent updates. So we should be ok, once we fix this.

**Comment by certik** [comment]
I have lowered the github update frequency, which fixes this problem. It's not so easy to make things better by accessing the store less, because we do want to have a time stamp at each record, so that we can see when it was last updated from github. So I think the solution is to update as we do now, and then implement a github hook to get updates whenever somebody creates a new pull request.

**Comment by certik** [comment]
As such, the only thing to do here is to keep trying if the upload fails.

**Comment by asmeurer** [comment]
Can we at least get a better error message?

**Comment by certik** [comment]
Ok, I am reopening it. Let's get a better error message.

**Comment by asmeurer** [comment]
This just happened again, and it didn't retry or anything.

**Comment by certik** [comment]
Ok, I checked what the problem was in the case of the issue #110 and we simply run over quota. So I just decreased the polling frequency from github even further (only once per day and twice per day respectively). Eventually we will have to redo how this is implemented.

**Comment by certik** [comment]
Please review the pull request #111 that fixes this issue by simply retrying until it gets uploaded.

**Comment by asmeurer** [comment]
Doesn't the polling not even work until we update it for API 3?

**Comment by certik** [comment]
You are right, maybe that's the reason. In any case, the most important thing is that the web app accepts the logs and that should work now.

**Comment by asmeurer** [comment]
#111 should have fixed this.
